syft/frameworks/torch/mpc/falcon.py

```diff
@@ -0,0 +1,82 @@
+import torch
+
+from syft.frameworks.torch.tensors.interpreters.replicated_shared import ReplicatedSharingTensor
+from syft.workers.base import BaseWorker
+
+from typing import List
+
+
+class Falcon(object):
+    @staticmethod
+    def _share_random_tensor(
+        field: int, workers: List[BaseWorker]
+    ) -> (ReplicatedSharingTensor, ReplicatedSharingTensor):
+        """
+        TODO: Need to move this in Primitive Store:
+        Share a random tensor with the owners of some shares
+        """
+        c = torch.randint(2, size=(1,))
+        c_share_L = c.share(*workers, protocol="falcon", field=field)
+        c_share_2 = c.share(*workers, protocol="falcon", field=2)
+        return c_share_2, c_share_L
+
+    @staticmethod
+    def select_share(
+        b_share: ReplicatedSharingTensor,
+        x_share: ReplicatedSharingTensor,
+        y_share: ReplicatedSharingTensor,
+    ) -> ReplicatedSharingTensor:
+        """ Performs select share protocol
+        If the bit b_share is 0, x_sh is returned
+        If the bit b_share is 1, y_sh is returned
+
+        Args:
+            x_share (ReplicatedSharingTensor): the first share to select
+            y_share (ReplicatedSharingTensor): the second share to select
+            b_share (ReplicatedSharingTensor): the bit to take into consideration
+                when selecting between x or y
+
+        Return:
+            x_share or b_share depending on the value of b_share
+        """
+
+        field = x_share.field
+        players = x_share.players
+        shape = x_share.shape
+        c_share_2, c_share_L = Falcon._share_random_tensor(field, players)
+
+        diff = y_share - x_share
+        xor_b_c = b_share.xor(c_share_2)
+
+        ones_share = ReplicatedSharingTensor.one_shares(players, field)
+        if xor_b_c.reconstruct().item():
+            d_share = ones_share - c_share_L
+        else:
+            d_share = c_share_L
+
+        selected_share = diff * d_share + x_share
```

Comment:
according to the paper the multiplication here should be done using \pi_mult (secure multiplication). 
I am not sure how this is suppose to work otherwise.